---
title: "P0_06052023_redo"
output: html_document
date: "2023-06-05"
---

this one was redone because the libraries were the same for 2 samples in the sequencing redone.
P0 was resequenced and sent on 05/30/2023
Cell Ranger was run and there were no warnings. Completed on Cheaha at 06/05/2023
Read Depth 29,386 reads per cell 
Number of Cells: 19,782
Valid barcodes: 98.2%
Valid UMIs: 100%

```{r}
library(dplyr)
library(Seurat)
library(patchwork)
```

```{r}
# Load the PBMC dataset
pbmc.data <- Read10X(data.dir =  "C:/Users/empage/Documents/Lab_Stuff/Anoikis/OV90/Single_Cell/P0_not_multiplexed_redo_06052023/cell_ranger_output/")
# Initialize the Seurat object with the raw (non-normalized data).
pbmc <- CreateSeuratObject(counts = pbmc.data, min.cells = 1, min.features = 200)
pbmc
```

```{r}
#add meta separately 
x2 <- rep("P0", each= 29582)
pbmc <- AddMetaData(object= pbmc, metadata = x2, col.name= "time")
```

```{r}
pbmc <- subset(pbmc, subset = nFeature_RNA > 200 & nFeature_RNA < 2500)

pbmc <- NormalizeData(pbmc, normalization.method = "LogNormalize", scale.factor = 10000)
```

```{r}
pbmc <- FindVariableFeatures(pbmc, selection.method = "mvp", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(pbmc), 10)

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(pbmc)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot1
```

```{r}
all.genes <- rownames(pbmc)
pbmc <- ScaleData(pbmc, features = all.genes)
```

```{r}
pbmc <- RunPCA(pbmc, features = VariableFeatures(object = pbmc))
```

```{r}
print(pbmc[["pca"]], dims = 1:20, nfeatures = 5)
```

```{r}
VizDimLoadings(pbmc, dims = 1:2, reduction = "pca")
```

```{r}
DimHeatmap(pbmc, dims = 3, cells = 500, balanced = TRUE)
```
#resolution 0.25 
```{r}
pbmc <- FindNeighbors(pbmc, dims = 1:40)
pbmc <- FindClusters(pbmc, resolution = 0.25)
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:3)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:15)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:20)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:35)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
pbmc.markers %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)

write.csv(pbmc.markers, "C:/Users/empage/Documents/Lab_Stuff/Anoikis/OV90/Single_Cell/Single_Cell_OV90_not_multi_UMAP_res_explore_062023/markers_res25_P0_dims35.csv")
```

#resolution 0.37
```{r}
pbmc <- FindNeighbors(pbmc, dims = 1:40)
pbmc <- FindClusters(pbmc, resolution = 0.37)
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:3)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:15)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:20)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:35)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
pbmc.markers %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)

write.csv(pbmc.markers, "C:/Users/empage/Documents/Lab_Stuff/Anoikis/OV90/Single_Cell/Single_Cell_OV90_not_multi_UMAP_res_explore_062023/markers_res37_P0_dims35.csv")
```

#resolution 0.5 
```{r}
pbmc <- FindNeighbors(pbmc, dims = 1:40)
pbmc <- FindClusters(pbmc, resolution = 0.5)
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:3)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:15)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:20)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:35)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
pbmc.markers %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)

write.csv(pbmc.markers, "C:/Users/empage/Documents/Lab_Stuff/Anoikis/OV90/Single_Cell/Single_Cell_OV90_not_multi_UMAP_res_explore_062023/markers_res50_P0_dims35.csv")
```

#Highest Group resolution 0.75
```{r}
pbmc <- FindNeighbors(pbmc, dims = 1:40)
pbmc <- FindClusters(pbmc, resolution = 0.75)
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:3)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:15)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:20)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc <- RunUMAP(pbmc, dims = 1:35)
DimPlot(pbmc, reduction = "umap")
```

```{r}
pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
pbmc.markers %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)

write.csv(pbmc.markers, "C:/Users/empage/Documents/Lab_Stuff/Anoikis/OV90/Single_Cell/Single_Cell_OV90_not_multi_UMAP_res_explore_062023/markers_res75_P0_dims35.csv")
```
